spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskExecutionService.java

```diff
@@ -339,6 +340,11 @@ else if(this.tasksBeingUpgraded.containsKey(taskName)) {
 
 		System.out.println(">> deployment props before launch: " + appDeploymentRequest.getDeploymentProperties());
 		String taskDeploymentId = taskLauncher.launch(appDeploymentRequest);
+
+		if(this.tasksBeingUpgraded.containsKey(taskName)) {
+			this.tasksBeingUpgraded.get(taskName).remove(platformName);
```

Comment:
If the list is now empty, we should remove the entry as well (to prevent a memory leak).  I can do this on merge.